ql/src/java/org/apache/hadoop/hive/ql/ddl/function/create/CreateFunctionOperation.java

```diff
@@ -116,11 +119,27 @@ private int createPermanentFunction() throws HiveException, IOException {
       return 1;
     }
 
-    boolean addToMetastoreSuccess = addToMetastore(dbName, functionName, registeredName);
-    if (!addToMetastoreSuccess) {
+    // TODO: should this use getUserFromAuthenticator instead of SessionState.get().getUserName()?
+    Function function = new Function(functionName, dbName, desc.getClassName(), SessionState.get().getUserName(),
```

Comment:
How security is handled for altering the functions?
Malicious users could wreak havoc if they change a function implementation behind the scenes

Reply:
Can we add `OWNER_PRIV` for altering function? Then only the owner of this function can do alter operation.